### PR TITLE
Schedule ViewModel

### PR DESCRIPTION
**Pull request by SidonieBouthors**
- [x] Show tasks on schedule
- [x] Filter shown tasks by selected events
- [x] Switch date by arrows functionality
- [x] Switch date by calendar

### CODE REVIEW

**Comment by Polymeth** [comment]
The schedule displayed is not correct for me. It's displayed 2 hours before. Probably a timeslot problem

File: app/src/main/java/com/github/se/assocify/ui/screens/event/scheduletab/EventScheduleViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import com.github.se.assocify.model.database.TaskAPI
+import com.github.se.assocify.model.entities.Event
+import com.github.se.assocify.model.entities.Task
+import com.github.se.assocify.ui.util.DateUtil
+import java.time.LocalDate
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class EventScheduleViewModel(
```
Comment (Polymeth) [inline app/src/main/java/com/github/se/assocify/ui/screens/event/scheduletab/EventScheduleViewModel.kt:11]: We really need to add documentation now, even to methods with straight forward names

Comment (SidonieBouthors) [reply app/src/main/java/com/github/se/assocify/ui/screens/event/scheduletab/EventScheduleViewModel.kt:13]: I added documentation :)

**Comment by SidonieBouthors** [comment]
I fixed the offset issue in the schedule :) the same issue was occurring in the task list so I corrected it globally and added a utils class to correctly deal with offset date times and displays

**Comment by Polymeth** [comment]
Perfect! Thanks for the work and the quick fixes :)

**Review by Polymeth — APPROVED** [review]

**Pull request merged**